packages/blade/src/components/Input/BaseInput/StyledBaseInput.web.tsx

```diff
@@ -55,36 +56,50 @@ const autoCompleteSuggestionTypeMap = {
   creditCardExpiryYear: 'cc-exp-year',
 };
 
-export const StyledBaseInput = ({
-  name,
-  isDisabled,
-  isRequired,
-  handleOnChange,
-  handleOnBlur,
-  keyboardReturnKeyType,
-  autoCompleteSuggestionType,
-  accessibilityProps,
-  setCurrentInteraction,
-  ...props
-}: StyledBaseInputProps): ReactElement => {
-  return (
-    <StyledBaseNativeInput
-      disabled={isDisabled}
-      required={isRequired}
-      onChange={(event): void => handleOnChange?.({ name, value: event })}
-      onBlur={(event): void => {
-        setCurrentInteraction('default');
-        handleOnBlur?.({ name, value: event });
-      }}
-      onFocus={(): void => setCurrentInteraction('focus')}
-      enterKeyHint={keyboardReturnKeyType}
-      autoComplete={
-        autoCompleteSuggestionType
-          ? autoCompleteSuggestionTypeMap[autoCompleteSuggestionType]
-          : undefined
-      }
-      {...props}
-      {...accessibilityProps}
-    />
-  );
-};
+export const StyledBaseInput = React.forwardRef<HTMLInputElement, StyledBaseInputProps>(
+  (
+    {
+      name,
+      isDisabled,
+      isRequired,
+      maxCharacters,
+      handleOnFocus,
+      handleOnChange,
+      handleOnBlur,
+      keyboardType,
+      keyboardReturnKeyType,
+      autoCompleteSuggestionType,
+      accessibilityProps,
+      setCurrentInteraction,
+      ...props
+    },
+    ref,
+  ) => {
+    return (
+      <StyledBaseNativeInput
+        ref={ref}
+        disabled={isDisabled}
+        required={isRequired}
+        maxLength={maxCharacters}
+        inputMode={keyboardType === 'telephone' ? 'tel' : keyboardType}
```

Comment:
nit: keep these `inputMode`, `enterKeyHint`, `autoComplete` after one another: 

```suggestion
        inputMode={keyboardType === 'telephone' ? 'tel' : keyboardType}
enterKeyHint={keyboardReturnKeyType === 'default' ? 'enter' : keyboardReturnKeyType}
autoComplete={
  autoCompleteSuggestionType
    ? autoCompleteSuggestionTypeMap[autoCompleteSuggestionType]
    : undefined
}
```

Reply:
fixed in 717f7a1da44ddbaf76edc37e72d60e182d95576b